### DISCUSSION TITLE
The various command-line options/GUI parameters would benefit from an explanation of when it's good/needed/possible to use/not use each

### DISCUSSION THREAD

**Discussion by TiagoTiago** [oobabooga/text-generation-webui#2388] [Q&A]
While a few things might be so obvious that even just the name already explain it all, other stuff even with the Description blurb still remain a mystery for anyone other than people that wouldn't need to check the documentation in the first place.

**Comment by ossirytk** [comment]
The main weakness of programmers and engineers is forgetting that not everyone is a programmer or an engineer. I've kind of realized that a lot of people are floundering in the dark and it's difficult to figure out how to do things when the information fragmented and scattered. I'll try to write a general documents in rentry or something when I find the time. 

Meanwhile, for llama.cpp the param explanations are in the readme.md in the examples/main/ folder. You can also find some references for run parameters and instructions in the HuggingFace model cards.

**Comment by melroy89** [comment]
It's a bit hidden in the [README](https://github.com/oobabooga/text-generation-webui). But try to click on " List of command-line flags" which should collapse the content and give you a list of available command line flags/parameters with description. Add a flag to `CMD_FLAGS.txt` file  (for example:  `--listen`) and run the script again.

Of course they could also create a executable/python script or whether with a `--help` command output. But I digress.

**Comment by araleza** [comment]
Got to say I cringed when I saw that the help text above the 'use_flash_attention_2' checkbox was 'Set use_flash_attention_2=True while loading the model.'  I mean, yeah, I know that, that doesn't help me at all.  The text should say _why_ I'd want to tick that box.  Will it speed up inference?  Or is it for faster LoRA Training?  Or if it mentioned some minimum compute capability my GPU would have to have, that would be useful too.  But instead it's just a bald repetition of the name of the flag.
